minecraft/networking/packets/clientbound/play/__init__.py

```diff
@@ -184,6 +201,27 @@ def get_id(context):
     ])
 
 
+class RespawnPacket(Packet):
+    @staticmethod
+    def get_id(context):
+        return 0x38 if context.protocol_version >= 389 else \
+               0x37 if context.protocol_version >= 352 else \
+               0x36 if context.protocol_version >= 345 else \
+               0x35 if context.protocol_version >= 336 else \
+               0x34 if context.protocol_version >= 332 else \
+               0x35 if context.protocol_version >= 318 else \
+               0x33 if context.protocol_version >= 70 else \
+               0x07
+
+    packet_name = 'respawn'
+    get_definition = staticmethod(lambda context: [
+        {'dimension': Integer},
+        {'difficulty': UnsignedByte},
+        {'game_mode': UnsignedByte},
+        {'level_type': String}
+    ])
```

Comment:
What you think to add a convenience subclass here that denotes the dimension/difficulty/game_mode?
The information could be shared with the [JoinGamePacket](https://github.com/ammaraskar/pyCraft/blob/master/minecraft/networking/packets/clientbound/play/__init__.py#L57)

```python
    class Difficulty(Enum):
        PEACEFUL = 0
        EASY = 1
        NORMAL = 2
        HARD = 3

    class Dimension(Enum):
        NETHER = -1
        OVERWORLD = 0
        END = 1

    class Gamemode(Enum):
        SURVIVAL = 0
        CREATIVE = 1
        ADVENTURE = 2
        SPECTATOR = 3
```

Reply:
I like this idea. Making them available as a type  in https://github.com/ammaraskar/pyCraft/blob/master/minecraft/networking/types/enum.py
would save us having to copy / paste the Enum class  in each packet definition...